packages/modules/payment/src/loaders/providers.ts

```diff
@@ -37,4 +43,32 @@ export default async ({
     providers: options?.providers || [],
     registerServiceFn: registrationFn,
   })
+
+  await registerProvidersInDb({ container })
+}
+
+const registerProvidersInDb = async ({
+  container,
+}: LoaderOptions): Promise<void> => {
+  const providersToLoad = container.resolve<string[]>("payment_providers")
+  const paymentProviderService = container.resolve<PaymentProviderService>(
+    "paymentProviderService"
+  )
+
+  const providers = await paymentProviderService.list({
+    id: providersToLoad,
+  })
+
+  const loadedProvidersMap = new Map(providers.map((p) => [p.id, p]))
+
+  const providersToCreate: CreatePaymentProviderDTO[] = []
+  for (const id of providersToLoad) {
+    if (loadedProvidersMap.has(id)) {
+      continue
+    }
+
+    providersToCreate.push({ id })
+  }
+
+  await paymentProviderService.create(providersToCreate)
```

Comment:
you can do an upsert with replace here to remove any providers that no longer exists.

Reply:
I also thought of it, but when I tried it it caused some issues in the platform (eg. if you've selected a default payment provider for the region, and then you remove it), so removing a payment provider will likely have to cascade to other modules as well, so it wasn't the right time :)